File: How_To_Program_Exercises/JavaHowToProgram9/Chapter3/EmployeeClass/Employee.java
//
//	Employee Class
//

class Employee
{
	private String firstName;
	private String lastName;
	private double monthlySalary;
	
	public Employee ( String fName, String lName, double monthlySal)
	{
		firstName = fName;
		lastName = lName;
		
		if ( monthlySal >= 0.0 )
			monthlySalary = monthlySal;
	}
	
	public String getFirstName()
	{
		return firstName;
	}
	
	public void setFirstName (String fName)
	{
		firstName = fName;
	}
	
	public String getLastName()
	{
		return lastName;
	}
	
	public void setLastName (String lName)
	{
		lastName = lName;
	}
	
	public double getMonthlySalary ()
	{
		return monthlySalary;
	}
	
	public void setMonthlySalary (double monthlySal)
	{
		if ( monthlySal > 0)
			monthlySalary = monthlySal;
	}
}

File: How_To_Program_Exercises/C++How_To_Program_8/Chapter3/EmployeeClass/Employee.cpp
//
//	Employee Class Implementation
//

#include <string>
#include "Employee.h"

Employee::Employee (std::string fName, std::string lName, int monthlySal)
{
	firstName = fName;
	lastName = lName;
	
	if (monthlySal > 0)
		monthlySalary = monthlySal;
}

std::string Employee::getFirstName()
{
	return firstName;
}

void Employee::setFirstName (std::string fName)
{
	firstName = fName;
}

std::string Employee::getLastName()
{
	return lastName;
}

void Employee::setLastName (std::string lName)
{
	lastName = lName;
}

int Employee::getMonthlySalary ()
{
	return monthlySalary;
}

void Employee::setMonthlySalary (int salary)
{
	if (salary > 0)
		monthlySalary = salary;
	else
		monthlySalary = 0;
}